Header header
DetectionMessage[] data
int32 detection_num